mitmproxy/addons/clientplayback.py

```diff
@@ -203,7 +203,7 @@ def start_replay(self, flows: typing.Sequence[flow.Flow]) -> None:
             # https://github.com/mitmproxy/mitmproxy/issues/2197
             if hf.request.http_version == "HTTP/2.0":
                 hf.request.http_version = "HTTP/1.1"
-                host = hf.request.headers.pop(":authority")
+                host = hf.request.headers.pop(":authority", hf.request.pretty_host)
```

Comment:
If there is no authority header (i.e. someone intentionally deleted it), I would argue we probably don't want a Host header in the replay either. How about we only add it if it exists, and do nothing otherwise?

Reply:
I'm really unsure about it as vhost selection will probably give strange results if `Host` header is missing (ie. giving you the default, or 1st in alphabetical order). 
But you're right on the no-guessing policy.